### PR TITLE
Add more ragger tests

### PR DESCRIPTION
**Pull request by fbeutin-ledger**
Hello,
I have finished ported the Solana application for our latest device Ledger Stax
My work is taking place on top of the swap modifications I made
https://github.com/LedgerHQ/app-solana/pull/44

I have separated the new feature in three independant PRs:

- Add more Ragger tests for automated non regression testing and Stax screen reproducability
https://github.com/LedgerHQ/app-solana/pull/61

- Extract the UI code from the logic handling code for the current graphical library BAGL
https://github.com/LedgerHQ/app-solana/pull/62

- And finally the Stax porting itself for all screens
https://github.com/LedgerHQ/app-solana/pull/59

### CODE REVIEW

File: .github/workflows/ci-workflow.yml [hunk 1]
```diff
@@ -10,75 +10,20 @@ on:
...
-          mkdir tests/elfs
-          mv *_binaries/*.elf tests/elfs
-      - name: Install APT dependencies
-        run: sudo apt-get update && sudo apt-get install -y qemu-user-static
-      - name: Install Ragger
-        run: |
-          pip install --extra-index-url https://test.pypi.org/simple/ -r tests/python/requirements.txt
-      - name: Run tests
-        run: pytest tests/python/ -v --tb=short --device all
+    uses: LedgerHQ/ledger-app-workflows/.github/workflows/reusable_ragger_tests.yml@v1
```
Comment (agrojean-ledger) [inline .github/workflows/ci-workflow.yml:22]: Don't you also want to add the rule enforcer job ?

File: tests/python/apps/solana_utils.py [hunk 2]
```diff
@@ -52,3 +56,81 @@ def lamports_to_bytes(lamports: int) -> str:
...
+               NavInsID.RIGHT_CLICK, NavInsID.RIGHT_CLICK, NavInsID.RIGHT_CLICK, NavInsID.BOTH_CLICK # Back to main menu
+              ]
+    else:
+        nav = [NavInsID.USE_CASE_HOME_SETTINGS,
+           NavIns(NavInsID.TOUCH, (348,132)),
+           NavInsID.USE_CASE_SETTINGS_MULTI_PAGE_EXIT]
+    navigator.navigate_and_compare(ROOT_SCREENSHOT_PATH,
+                                   snapshots_name,
+                                   nav,
+                                   screen_change_before_first_instruction=False)
```
Comment (xchapron-ledger) [inline tests/python/apps/solana_utils.py:75]: There is a bit of Stax related code here, but don't bother changing it, It's just forward test implementation following the concept of creating test before coding the functionality!

**Review by xchapron-ledger — APPROVED** [review]

File: tests/python/test_solana.py [hunk 3]
```diff
@@ -1,67 +1,227 @@
...
-                                                  test_name)
+        with sol.send_public_key_with_confirm(SOL_PACKED_DERIVATION_PATH):
+            if backend.firmware.device.startswith("nano"):
+                navigator.navigate_until_text_and_compare(NavInsID.RIGHT_CLICK,
+                                                          [NavInsID.BOTH_CLICK],
+                                                          "Approve",
+                                                          ROOT_SCREENSHOT_PATH,
+                                                          test_name)
+            else:
+                navigator.navigate_and_compare(ROOT_SCREENSHOT_PATH,
```
Comment (xchapron-ledger) [inline tests/python/test_solana.py:25]: I find this one weird. Did you use nbgl_useCaseReviewStart and nbgl_useCaseAddressConfirmation as in the guidelines?

**Review by agrojean-ledger — DISMISSED** [review]

File: tests/python/test_solana.py [hunk 4]
```diff
@@ -1,67 +1,227 @@
...
+                                               test_name,
+                                               [NavInsID.USE_CASE_ADDRESS_CONFIRMATION_CONFIRM])
```
Comment (xchapron-ledger) [inline tests/python/test_solana.py:27]: I thin you should add `NavInsID.USE_CASE_STATUS_DISMISS` at the end of all stax flows so thatt they are correctly ended.

File: .github/workflows/ci-workflow.yml (hunk 1 above)
Comment (fbeutin-ledger) [reply .github/workflows/ci-workflow.yml:22]: I'll do it when it's all merged, too many branches on top of branches

File: tests/python/test_solana.py (hunk 3 above)
Comment (fbeutin-ledger) [reply tests/python/test_solana.py:25]: Nope... Good catch

**Pull request merged**